### PR TITLE
refactor(language-service): move to using mockHost as much as possible

### PR DESCRIPTION
**Pull request by ayazhafiz**
Update tests that still do not use `mockHost` for certain operations,
like `addCode`.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/language-service/test/completions_spec.ts [hunk 1]
```diff
@@ -202,14 +202,11 @@ export class MyComponent {
         });
   });
 
-  function addCode(code: string, cb: (fileName: string, content?: string) => void) {
-    const fileName = '/app/app.component.ts';
-    const originalContent = mockHost.getFileContent(fileName);
-    const newContent = originalContent + code;
-    mockHost.override(fileName, originalContent + code);
+  function addCodeAndCallback(code: string, cb: (fileName: string, content?: string) => void) {
```
Comment (kyliau) [inline packages/language-service/test/completions_spec.ts]: We could get rid of this helper function entirely. There's no need for callback since `mockHost.addCode()` now returns the fileName. The callback is mainly to "reset" the mockHost. We could do that in a `beforeEach()` instead.
To keep things moving, we could do it in another PR, it's up to you.

**Review by kyliau — APPROVED** [review]

File: packages/language-service/test/completions_spec.ts (hunk 1 above)
Comment (ayazhafiz) [reply packages/language-service/test/completions_spec.ts]: I'll fix this in another PR; good call.
